src/script/common/c_content.cpp

```diff
@@ -2053,8 +2058,9 @@ void push_hud_element(lua_State *L, HudElement *elem)
 	lua_setfield(L, -2, "number");
 
 	if (elem->type == HUD_ELEM_WAYPOINT) {
-		// waypoints reuse the item field to store precision, precision = item - 1
-		lua_pushnumber(L, elem->item - 1);
+		// Waypoints reuse the item field to store precision,
+		// item = precision + 1 and item = 0 <=> precision = 10 for backwards compatibility.
+		lua_pushnumber(L, (elem->item == 0) ? 10 : (elem->item - 1));
```

Comment:
Please refer to `Hud::drawLuaElements` to hopefully stay in sync and to understand the origin of this formula.